### PR TITLE
refactor: remove dependency on AnotherWorld

### PR DESCRIPTION
**Pull request by naalit**
GrowingFlora currently depends on AnotherWorld, but there isn't much reason for that to be the case. I moved some utilities here and to the engine (see https://github.com/MovingBlocks/Terasology/pull/4849, which this PR depends on) and updated some references, which was enough to remove the dependency. Behavior shouldn't change, except that it no longer considers parent biomes for `AnotherWorldsBiome`s, and it also no longer has a bug where branch angles were snapped to whole numbers of radians.

Eventually, it would be nice to remove the dependency on ClimateConditions (or hopefully make it optional so that climate still affects growth if it is present), since Metal Renegades would like to use GrowingFlora but doesn't necessarily want to enable ClimateConditions.

### CODE REVIEW

File: src/main/java/org/terasology/gf/generator/FoliageProvider.java [hunk 1]
```diff
@@ -19,28 +20,29 @@
...
 
-        for (Map.Entry<Vector3i, Float> positionValue : floraFacet.getFlaggedPositions().entrySet()) {
-            Vector3i pos = positionValue.getKey();
+        for (Map.Entry<Vector3ic, Float> positionValue : floraFacet.getWorldEntries().entrySet()) {
+            Vector3ic pos = positionValue.getKey();
             float value = positionValue.getValue();
 
-            if (noise.noise(pos.x, pos.y, pos.z) / 256f < amount) {
-                facet.setFlag(pos, value);
+            if (foliageRegion.contains(pos) && noise.noise(pos.x(), pos.y(), pos.z()) / 256f < amount) {
```
Comment (skaldarnar) [inline src/main/java/org/terasology/gf/generator/FoliageProvider.java:44]: Is there no variant to pass a vector to `noise.noise(pos)`? If that is missing, we should add it...

File: src/main/java/org/terasology/gf/PlantGrowingSystem.java [hunk 2]
```diff
@@ -64,7 +65,7 @@ public void updatePlant(DelayedActionTriggeredEvent event, EntityRef plant, Livi
             PerformanceMonitor.startActivity("GrowingFlora - Updating plant");
             try {
                 PlantGrowthDefinition plantDefinition = plantRegistry.getPlantGrowthDefinition(plantComponent.type);
-                Long updateDelay = plantDefinition.requestedUpdatePlant(worldProvider, environmentSystem, blockEntityRegistry, plant);
+                Long updateDelay = plantDefinition.requestedUpdatePlant(worldProvider, new EnvironmentLocalParameters(environmentSystem, blockComponent.getPosition()), blockEntityRegistry, plant);
```
Comment (skaldarnar) [inline src/main/java/org/terasology/gf/PlantGrowingSystem.java]: This whole system will only be initialized if all it's dependencies can be met. If ClimateConditions is not loaded, we'll see logs like the following, and the whole feature is not present. 

```
22:47:03.042 [main] WARN  o.t.e.core.ComponentSystemManager - Skip system GrowingFlora:SaplingInitializeSystem for loading - possibly missing optional dependencies
22:47:03.045 [main] WARN  o.t.e.core.ComponentSystemManager - Skip system GrowingFlora:PlantGrowingSystem for loading - possibly missing optional dependencies
```

I think this is fine for now, but we can definitely improve on this, e.g., by basing those plant definition on something else if the `environmentSystem` is not present.

I think this is something for another PR, though. Maybe Terasology/SimpleFarming#120 can give some reference.

**Review by skaldarnar — DISMISSED** [review]
Can confirm this works with https://github.com/Terasology/MetalRenegades/pull/161

File: src/main/java/org/terasology/gf/PlantGrowingSystem.java (hunk 2 above)
Comment (naalit) [reply src/main/java/org/terasology/gf/PlantGrowingSystem.java]: I actually fixed this right before this comment. Those systems are needed for plants to grow, so skipping them isn't great.

File: src/main/java/org/terasology/gf/generator/FoliageProvider.java (hunk 1 above)
Comment (naalit) [reply src/main/java/org/terasology/gf/generator/FoliageProvider.java:44]: No, there isn't. Should that be added to the main `Noise` interface?

Comment (skaldarnar) [reply src/main/java/org/terasology/gf/generator/FoliageProvider.java:44]: I think that makes sense, yes. We can use a default implementation for that to only touch that one file.

**Pull request merged**